### DISCUSSION TITLE
What is best location to get locales / internationalization files?

### DISCUSSION THREAD

**Discussion by f1-outsourcing** [OpenMage/magento-lts#1950] [Questions & Answers] [answered]
Where can I find the most up to date archive for localization files for magento? These[1] are quite old and not complete. Maybe create a repository for this on OpenMage, so they can be maintained here?

[1]
https://github.com/versedi/Magento-Locales

**Comment by luigifab** [comment]
Sadly, there are no other places. You can found elsewhere some translation packs.
I will think to PR a cleaned version of _versedi/Magento-Locales_, because it's a must have.

**Reply by f1-outsourcing** [reply]
Would be great to have it here. I am currently seeing quite a few weird translations so it is time to up date them.

**Reply by luigifab** [reply]
There are also https://crowdin.com/project/magento-1 and https://crowdin.com/project/magento-2.

**Reply by luigifab** [reply]
If I understand well, this is AFL/OSL v3: https://github.com/magento-l10n/language-fr_FR

**Reply by luigifab** [reply]
So we can get many recent translations. Yes?

**Comment by aterjung** [comment]
I would like to see a OM repo for language packs. For these language packs i know well, there is no more development done.
But i am not shure if https://github.com/versedi/Magento-Locales is a good source. It looks very outdated. Probably we should start from scratch and only add translations known to be good? Its impossible to check a translations quality if you don't speek the language.
At least, we should check the license for the language-packs.

**Reply by luigifab** [reply]
Yes, there are tons of mistakes from these translations (I have a script to fix CSV syntax and encoding). I using it, and have fixed some of them. I also mixed it from translations packs found elsewhere. From memory, there are 8000 strings to translate, and I prefer to start from existing translations.

From web archive, there are no license information:
![image](https://user-images.githubusercontent.com/31816829/148534830-b8cab378-efb5-4e14-a8da-a3f140448b0e.png)

Not sure how to found the information. Magento is dual licensed OSL/AFL v3, perhaps these translations use same licences. Perhaps we can ask at:
https://github.com/OpenMage/magento-lts/blob/f99c9b031f4d3d6d06d7770f11cad1c7fa38456f/index.php#L13

**Reply by luigifab** [reply]
I remember that some CSV files from _versedi/Magento-Locales_ are wrong (something like _es_ES_ in _de_DE_).
Perhaps we can increase the percent for each locales by using open source languages packs from Magento 2 (not searched).

**Reply by addison74** [reply]
These translations are old and outdated in my opinion. Most likely many strings are missing from them. It would be ideal to start with the existing ones in OpenMage.

Speaking of translations, I remember meeting a developer who had concatenated all CSV translation files into one file named **translate.csv**. He had removed the duplicates and made the changes quickly. I have never considered this option because I am not in a similar situation, but maybe it would not be a bad idea at all.

**Reply by luigifab** [reply]
There are no translations in OpenMage, only _en_US_. In my previous comment I take all the CSV files from _versedi/Magento-Locales_, then I rebuild them by using _en_US_ CSV files from OpenMage.

Final CSV are clean.

**Comment by luigifab** [comment]
Counts updated: https://github.com/luigifab/openmage-translations

**Reply by f1-outsourcing** [reply]
Magento2 preferred above versedi?

**Reply by luigifab** [reply]
Yes! Translations from Magento2 are preferred.

**Reply by addison74** [reply]
@luigifab - Your work is to be appreciated. A separate repository should be created where all these translations should exist. Then it would be great to have the contribution of as many native speakers on each language as possible through PRs.

**Reply by luigifab** [reply]
Yes, there are many work to do.

**Reply by f1-outsourcing** [reply]
Maybe it is good to have a different system for editing these locales instead of only working with csv files. Just for the purpose of having a better control of editing them. With gettext for instance you have some 'tools' to manage translations files.
Such a system should also take into account that one locale can have a different versions like a more casual and a more professional version, maybe also addressing.
Also maybe a little manual for when people add different strings to a theme or so, how they should work with translations, so these custom translations are not ending up in the repository.

**Reply by f1-outsourcing** [reply]
@luigifab 
> Yes, there are many work to do.

Can you create a repo so I can fork it and work on some translations?

**Reply by luigifab** [reply, marked as answer]
I created a [repo](https://github.com/luigifab/openmage-translations).
I imported data from Magento2, riconeitzel and versedi.
Fully not tested and not checked.

**Reply by f1-outsourcing** [reply]
It looks like you currently left out 'key strings', if they do not have a value. I guess if you put a value of "" it will just blank the translation. But is this also when you do something like this:

`Mage_Checkout.csv:"Checkout Method",`

Because it would be handy if untranslated keys are already in these files, and you do not have to look them up. Unless there is some tool that can help with this?

**Reply by f1-outsourcing** [reply]
And would you like to have additions to these files at the end or first ordered like they are now?

**Comment by aterjung** [comment]
For German, the de-facto standard is:

https://github.com/riconeitzel/German_LocalePack_de_DE

Aren’t there similar projects for the other locales?
